src/node_graph/group.py

```diff
@@ -0,0 +1,22 @@
+class Group:
```

Comment:
The name `Group` is too general, looking at the code and the docstring, maybe change to `DependencyGroup`.